pkg/webhook/node/plugins/resourceamplification/resource_amplification.go

```diff
@@ -0,0 +1,162 @@
+		}
+	}
+	return !util.IsResourceListEqual(oldOriginalResources, newOriginalResources)
+}
+
+func (n *NodeResourceAmplificationPlugin) handleUpdate(oldNode, node *corev1.Node) error {
+	if node.Annotations[extension.AnnotationNodeResourceAmplificationRatio] == "" {
+		// When the feature is turned off, the saved raw allocatable needs to be cleaned up
+		delete(node.Annotations, extension.AnnotationNodeRawAllocatable)
+		return nil
+	}
+	if node.Status.Allocatable == nil {
+		// do nothing
+		return nil
+	}
+
+	// Set original resource under two conditions
+	// 1. if the original resource is not set
+	// 2. Node allocatable are updated, such as the administrator adjusting the reserved resource
+	//    https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources
+	//    Here, we rely on an assumption that only kubelet will update allocatable, and other roles will only expand new fields and not overwrite native fields.
+	//    FIXME If the updated resource of kubelet happens to be the same as the amplificated allocatable, the update will not take effect.
+	originalResources := corev1.ResourceList{}
+	if !extension.HasNodeRawAllocatable(node.Annotations) || n.isSupportedResourceChanged(oldNode, node) {
+		// resources in other dimensions will be dynamically updated, so they are not saved here. The caller needs to be aware of this feature.
+		for _, resourceName := range supportedResources {
+			resourceValue, ok := node.Status.Allocatable[resourceName]
+			if ok {
+				originalResources[resourceName] = resourceValue
+			}
+		}
+		if len(originalResources) > 0 {
+			klog.V(2).Infof("admit for %s, set raw allocatable %v", node.Name, originalResources)
+			extension.SetNodeRawAllocatable(node, originalResources)
+		}
+	} else {
+		originalResources, err := extension.GetNodeRawAllocatable(node.Annotations)
+		if originalResources == nil {
+			klog.Errorf("admit for %s, failed to parse raw allocatable: %v", node.Name, err)
+			return nil
+		}
+	}
+
+	// parse resource amplification ratios from annotations
+	amplificationRatios, err := extension.GetNodeResourceAmplificationRatios(node.Annotations)
+	if err != nil {
+		klog.Errorf("admit for %s, failed to parse resource amplification ratios: %v", node.Name, err)
+		return nil
+	}
+
+	// update allocatable by originalResources * amplificationRatios:
+	//     originalResources: {cpu: 1000, memory: 2000} * amplificationRatios: {cpu: 2, memory: 3} = {cpu: 2000, memory: 6000}
+	// If the resources saved in original are missing, the original will still prevail, and the missing dimensions will not be scaled:
+	//     originalResources: {cpu: 1000} * amplificationRatios: {cpu: 2, memory: 3} = {cpu: 2000}
+	// If the resource dimensions supported by the amplification change, the latest one shall prevail, and the missing dimensions will not be amplified:
+	//     originalResources: {cpu: 1000, memory: 2000} * amplificationRatios: {cpu: 2} = {cpu: 2000}
```

Comment:
This should be `originalResources: {cpu: 1000, memory: 2000} * amplificationRatios: {cpu: 2} = {cpu: 2000, memory: 2000}`?

Reply:
Only the amplified results are shown here, not the final results. The final result is obtained by updating the allocatable with the amplified result.